src/main.py

```diff
@@ -1,17 +1,28 @@
-from flask import Flask, request
-
+from flask import Flask, request, Response , json 
+from store.sparql import SparqlStore
 
 app = Flask(__name__)
+storage = SparqlStore()
+jsonld = "application/ld+json"
 
 @app.route("/datasets", methods =["GET"])
 def datasets():
-    if request.headers['Accept'] == "application/ld+json":
-        return get_json(request)
+    status_code = 200
+    if request.headers['Accept'] == jsonld: 
+        result = storage.get_datasets()
+        
+        if len(result["items"]) == 0:
+            status_Code  = 404
+        
+        resp = Response(json.dumps(result))
```

Comment:
there's a flask "jsonify" function for returning json objects, probably worth a quick investigation: https://www.geeksforgeeks.org/use-jsonify-instead-of-json-dumps-in-flask/